packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1513 @@
+      };
+
+      for (const doc of agreementConsumerDocuments) {
+        await uploadDocument(agreementId, doc.id, doc.name);
+      }
+
+      const agreementToBeUpgraded: Agreement = {
+        ...getMockAgreement(
+          generateId<EServiceId>(),
+          consumerId,
+          randomArrayItem(agreementUpgradableStates)
+        ),
+        id: agreementId,
+        descriptorId,
+        producerId,
+        stamps: {},
+        createdAt: TEST_EXECUTION_DATE,
+        consumerDocuments: agreementConsumerDocuments,
+      };
+
+      await addOneAgreement(agreementToBeUpgraded, postgresDB, agreements);
+
+      const eservice: EService = getMockEService(
+        agreementToBeUpgraded.eserviceId,
+        tenant.id,
+        [deprecatedDescriptor, publishedDescriptor]
+      );
+      await addOneEService(eservice, eservices);
+
+      const newAgreementId = unsafeBrandId<AgreementId>(
+        await agreementService.upgradeAgreement(
+          agreementToBeUpgraded.id,
+          authData,
+          uuidv4()
+        )
+      );
+
+      expect(newAgreementId).toBeDefined();
+      const actualAgreementCreatedEvent = await readAgreementEventByVersion(
+        newAgreementId,
+        0,
+        postgresDB
+      );
+
+      expect(actualAgreementCreatedEvent).toMatchObject({
+        type: "AgreementAdded",
+        event_version: 1,
+        version: "0",
+        stream_id: newAgreementId,
+      });
+
+      const actualCreatedAgreement = decodeProtobufPayload({
+        messageType: AgreementAddedV1,
+        payload: actualAgreementCreatedEvent.data,
+      }).agreement;
+
+      const expectedCreatedAgreement = toAgreementV1({
+        id: newAgreementId,
+        eserviceId: agreementToBeUpgraded.eserviceId,
+        descriptorId,
+        producerId: agreementToBeUpgraded.producerId,
+        consumerId: agreementToBeUpgraded.consumerId,
+        verifiedAttributes: agreementToBeUpgraded.verifiedAttributes,
+        certifiedAttributes: agreementToBeUpgraded.certifiedAttributes,
+        declaredAttributes: agreementToBeUpgraded.declaredAttributes,
+        consumerNotes: agreementToBeUpgraded.consumerNotes,
+        state: agreementState.draft,
+        createdAt: TEST_EXECUTION_DATE,
+        consumerDocuments: [],
+        stamps: {},
+      });
+
+      expectedCreatedAgreement.stamps = {};
+      delete expectedCreatedAgreement.updatedAt;
+      delete expectedCreatedAgreement.suspendedAt;
+      delete expectedCreatedAgreement.contract;
+      expect(actualCreatedAgreement).toMatchObject(expectedCreatedAgreement);
+
+      // The following assertions verify the "AgreementConsumerDocumentAdded" event for each document
+      for (let index = 0; index < agreementConsumerDocuments.length; index++) {
+        const agreementConsumerDocument = agreementConsumerDocuments[index];
+        const currentVersion = index + 1;
+        const actualAgreementDocumentAddedEvent =
+          await readEventByStreamIdAndVersion(
```

Comment:
Use `readAgreementEventByVersion` from the agreement test `utils.ts` instead of the one from commons-test

Reply:
✅  done in commit [f049179](https://github.com/pagopa/interop-be-monorepo/pull/392/commits/f049179b3b3e8e60614f84c5b46bfa8f7c793ed8)